### PR TITLE
td-shim: follow the td-shim trust boot guideline

### PR DESCRIPTION
**Pull request by gaojiaqi7**
- Record the TCG_EfiSpecIDEvent as the first event.

 - For TD HOB, extend it to RTMR[0] and record the event with event type EV_PLATFORM_CONFIG_FLAGS and event data being TD_SHIM_PLATFORM_CONFIG_INFO.

 - For payload binary, extend it to RTMR[1] and record the event with event type EV_EFI_PLATFORM_FIRMWARE_BLOB2 and event data being UEFI_PLATFORM_FIRMWARE_BLOB2.

 - For secure boot, extend the provisioned and matched trust anchor into RTMR[0] and payload SVN into RTMR[1] and record them with event type EV_PLATFORM_CONFIG_FLAGS and event data being TD_SHIM_PLATFORM_CONFIG_INFO.

 - Before jumping to the TD payload, td-shim measures 0x0000_0000 and record the EV_SEPARATOR into the event log to mark the end of the td-shim events.

### CODE REVIEW

File: td-shim/src/bin/td-shim/tcg.rs [hunk 1]
```diff
@@ -34,7 +42,20 @@ impl TdEventLog {
...
+        };
+
+        // Create the TCG_EfiSpecIDEvent as the first event
+        let first = TcgEfiSpecIdevent::default();
+        td_event_log
+            .log_event(
+                0,
+                EV_NO_ACTION,
+                first.as_bytes(),
+                &[0u8; SHA384_DIGEST_SIZE],
```
Comment (jyao1) [inline td-shim/src/bin/td-shim/tcg.rs:54]: The digest for SpecidEvent is always 20 bytes - to align with SHA1 for compatibility reason.

File: td-shim/src/event_log.rs [hunk 2]
```diff
@@ -16,10 +19,143 @@ pub const PCR_DIGEST_NUM: usize = 1;
...
+
+    pub fn as_bytes(&self) -> &[u8] {
+        unsafe { &*slice_from_raw_parts(self as *const Self as *const u8, size_of::<Self>()) }
+    }
+}
+
+impl Default for TcgEfiSpecIdevent {
+    fn default() -> Self {
+        Self {
+            signature: *b"Spec ID Event01\0",
```
Comment (jyao1) [inline td-shim/src/event_log.rs:79]: Why you use "01", the PFP spec says "03", refer to Table 20 TCG_EfiSpecIdEvent.

File: td-shim/src/event_log.rs [hunk 3]
```diff
@@ -16,10 +19,143 @@ pub const PCR_DIGEST_NUM: usize = 1;
...
+            platform_class: 0,
+            spec_version_minor: 0,
+            spec_version_major: 0x2,
+            spec_errata: 0,
```
Comment (jyao1) [inline td-shim/src/event_log.rs:83]: errata is 105

File: td-shim/src/event_log.rs (hunk 2 above)
Comment (gaojiaqi7) [reply td-shim/src/event_log.rs]: Yes it should be '03', misunderstood the content and thought it is an example

**Pull request merged**